thirdparty/github.go

```diff
@@ -497,82 +474,68 @@ func GithubAuthenticate(code, clientId, clientSecret string) (githubResponse *Gi
 	if err != nil {
-		return nil, nil, ResponseReadError{err.Error()}
-	}
-
-	grip.Debugf("Github API response: %s. %d bytes", resp.Status, len(respBody))
-
-	if err = json.Unmarshal(respBody, &githubUser); err != nil {
-		return nil, nil, APIUnmarshalError{string(respBody), err.Error()}
+		return nil, false, errors.Wrap(err, "can't fetch data from github")
 	}
+	defer util.PutHttpClientForOauth2(httpClient)
+	client := github.NewClient(httpClient)
 
-	// get the user's organizations
-	resp, err = tryGithubGet(t, orgUrl)
-	if resp != nil {
-		defer resp.Body.Close()
-	}
-	if err != nil {
-		return nil, nil, errors.Wrapf(err, "Could not get user from token")
-	}
-	respBody, err = ioutil.ReadAll(resp.Body)
+	user, resp, err := client.Users.Get(context.TODO(), "")
 	if err != nil {
-		return nil, nil, ResponseReadError{err.Error()}
+		return nil, false, errors.WithStack(err)
 	}
-
-	grip.Debugf("Github API response: %s. %d bytes", resp.Status, len(respBody))
-
-	if err = json.Unmarshal(respBody, &githubOrganizations); err != nil {
-		return nil, nil, APIUnmarshalError{string(respBody), err.Error()}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		var respBody []byte
+		respBody, err = ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, false, ResponseReadError{err.Error()}
+		}
+		return nil, false, APIResponseError{string(respBody)}
 	}
-	return
-}
 
-// verifyGithubAPILimitHeader parses a Github API header to find the number of requests remaining
-func verifyGithubAPILimitHeader(header http.Header) (int64, error) {
-	h := (map[string][]string)(header)
-	limStr, okLim := h["X-Ratelimit-Limit"]
-	remStr, okRem := h["X-Ratelimit-Remaining"]
-
-	if !okLim || !okRem || len(limStr) == 0 || len(remStr) == 0 {
-		return 0, errors.New("Could not get rate limit data")
+	var isMember bool
+	if len(requiredOrg) > 0 {
+		isMember, _, err = client.Organizations.IsMember(context.TODO(), requiredOrg, *user.Login)
+		if err != nil {
+			return nil, false, errors.Wrapf(err, "Could check if user was org member")
+		}
 	}
 
-	rem, err := strconv.ParseInt(remStr[0], 10, 0)
-	if err != nil {
-		return 0, errors.Errorf("Could not parse rate limit data: limit=%q, rate=%t", limStr, okLim)
+	if user.Login == nil || user.ID == nil || user.Company == nil ||
+		user.Email == nil || user.OrganizationsURL == nil {
+		return nil, false, errors.New("Github user is missing required data")
 	}
 
-	return rem, nil
+	return &GithubLoginUser{
+		Login:            *user.Login,
+		Id:               *user.ID,
+		Company:          *user.Company,
+		EmailAddress:     *user.Email,
+		OrganizationsURL: *user.OrganizationsURL,
+	}, isMember, err
 }
 
 // CheckGithubAPILimit queries Github for the number of API requests remaining
-func CheckGithubAPILimit(token string) (int64, error) {
-	url := fmt.Sprintf("%v/rate_limit", GithubAPIBase)
-	resp, err := githubRequest("GET", url, token, nil)
+func CheckGithubAPILimit(oauthToken string) (int64, error) {
```

Comment:
pass context into this?

Reply:
additionally I think this now diverges from the functional similar getGithubRateLimit function used elsewhere

Reply:
getGithubRateLimit creates a string message based on the rate limit returned from API responses. This function should have been calling and logging that data though.